kge/job/job.py

```diff
@@ -55,36 +58,115 @@ def __init__(self, config: Config, dataset: Dataset, parent_job: "Job" = None):
-        """Load job state from last or specified checkpoint.
+    @staticmethod
+    def create(
+        config: Config, dataset: Optional[Dataset] = None, parent_job=None, model=None
+    ):
+        from kge.job import TrainingJob, EvaluationJob, SearchJob
 
-        Restores all relevant state to resume a previous job. To run the restored job,
-        use :func:`run`.
+        if dataset is None:
+            dataset = Dataset.load(dataset)
+
+        job_type = config.get("job.type")
+        if job_type == "train":
+            return TrainingJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        elif job_type == "search":
+            return SearchJob.create(config, dataset, parent_job=parent_job)
+        elif job_type == "eval":
+            return EvaluationJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        else:
+            raise ValueError("unknown job type")
 
-        Should set `resumed_from_job` to the job ID of the previous job.
+    @staticmethod
+    def find_and_create_from(
+        checkpoint: str = None,
+        config: Config = None,
+        dataset: Dataset = None,
+        parent_job=None,
+    ) -> Job:
+        """
+        Finds and loads the checkpoint to resume from given a config file in in the
+        experiment folder structure.
+        If a checkpoint is specified, it will be loaded. In this case no config file
+        needs to be specified.
+        Args:
+            checkpoint: path to checkpoint file
+            config: config in the experiment folder structure
+            dataset: dataset object
+            parent_job: parent job (e.g. search job)
+
+        Returns: Job based on checkpoint or new Job if no checkpoint found
 
         """
-        raise NotImplementedError
+        if checkpoint is None and config is None:
+            raise ValueError(
+                "Please provide either the config file located in the folder structure "
+                "containing the checkpoint or the checkpoint itself."
+            )
+        elif checkpoint is None:
+            last_checkpoint = config.last_checkpoint()
+            if last_checkpoint is not None:
+                checkpoint = config.checkpoint_file(last_checkpoint)
+
+        if checkpoint is not None:
+            job = Job.create_from(checkpoint, config, dataset, parent_job=parent_job)
+            if type(checkpoint) == str:
+                job.config.log("Loading checkpoint from {}...".format(checkpoint))
+        else:
+            job = Job.create(config, dataset, parent_job=parent_job)
+            job.config.log("No checkpoint found, starting from scratch...")
+        return job
 
-    def run(self):
-        raise NotImplementedError
+    @classmethod
+    def create_from(
+        cls,
+        checkpoint: Union[str, Dict],
+        config: Config = None,
+        dataset: Dataset = None,
+        parent_job=None,
```

Comment:
Is it really required to have create_from and find_and_create_from and create?

Reply:
we could merge create_from and find_and_create_from, but then check, but then the checkpoint parameter would be Optional[Union[str, Dict]].